### PR TITLE
Add tag to jenkins_common to allow local dev

### PR DESCRIPTION
**Pull request by michaelyoungstrom**
Creates a tag that we can run for local dev (so we dont need to run all the services like nginx/ splunkforwarder/ etc). It also includes new stanzas for deleting existing files to avoid any unwanted configuration.

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

File: playbooks/roles/jenkins_common/tasks/main.yml [hunk 1]
```diff
@@ -132,17 +150,28 @@
...
     - install
     - install:base
     - install:jenkins-configuration
 
+- name: Copy all init scripts other than oauth for local dev
+  command: 'cp {{ jenkins_common_git_home }}/jenkins-configuration/{{ jenkins_common_configuration_src_path }}/{{ item }} {{ jenkins_common_home }}/init.groovy.d/'
+  with_items: '{{ jenkins_common_configuration_scripts }}'
+  become: true
+  become_user: '{{ jenkins_common_user }}'
+  when: 'item != "4configureGHOAuth.groovy" and init_scripts_copied is not defined'
```
Comment (estute) [inline playbooks/roles/jenkins_common/tasks/main.yml:171]: I feel like we will want to split that script up into an Oauth script and a security script

**Review by estute — COMMENTED** [review]
I approve the changes, but am curious about some other things. What about:
1) how will this handle nginx in playbooks/roles/jenkins_common/meta.yml?
2) how will this handle splunk and data dog configured in playbooks/edx-east/jenkins_build.yml? They shouldn't be running in a container
3) will we run into any trouble by running the aws pla in playbooks/edx-east/jenkins_build.yml?

**Comment by michaelyoungstrom** [comment]
@estute None of those are run when configuring the container with this new tag. To be clear, here's what gets executed by the tag:

```
    tasks:
      jenkins_common : Delete any existing jenkins-configuration folders to avoid unwanted configuration	TAGS: [install, install:base, install:jenkins-configuration, jenkins:local-dev]
      jenkins_common : Create necessary folders	TAGS: [install, install:base, install:jenkins-configuration, jenkins:local-dev]
      jenkins_common : Download jenkins-configuration repo	TAGS: [install, install:base, install:jenkins-configuration, jenkins:local-dev]
      jenkins_common : Run gradle libs	TAGS: [install, install:base, install:jenkins-configuration, jenkins:local-dev]
      jenkins_common : Copy all init scripts other than oauth for local dev	TAGS: [jenkins:local-dev]
      jenkins_common : Create jenkins config sub folders	TAGS: [install, install:base, install:jenkins-configuration, jenkins:local-dev]
      jenkins_common : For local dev, copy any config files other than oauth	TAGS: [jenkins:local-dev]
      jenkins_common : Copy plugins.yml config file	TAGS: [install, install:base, install:jenkins-configuration, install:plugins, jenkins:local-dev]
      jenkins_common : Copy ec2 config files	TAGS: [install, install:base, install:jenkins-configuration, jenkins:local-dev]
      jenkins_common : Copy xml config files	TAGS: [install, install:base, install:jenkins-configuration, jenkins:local-dev]
      jenkins_common : Run plugins.gradle	TAGS: [install, install:base, install:jenkins-configuration, install:plugins, jenkins:local-dev]
      jenkins_common : Copy secret file credentials	TAGS: [install, install:base, install:jenkins-configuration, jenkins:local-dev]
      jenkins_common : Copy ssh key credentials	TAGS: [install, install:base, install:jenkins-configuration, jenkins:local-dev]
      jenkins_common : Copy ec2 key	TAGS: [install, install:base, install:jenkins-configuration, jenkins:local-dev]
```

**Review by estute — APPROVED** [review]
:+1:

File: playbooks/roles/jenkins_common/tasks/main.yml (hunk 1 above)
Comment (michaelyoungstrom) [reply playbooks/roles/jenkins_common/tasks/main.yml:171]: Noted. I'll add a ticket to the backlog regarding this.

**Pull request merged**